### PR TITLE
Creating a virtual assistant using python

### PR DESCRIPTION
**Pull request by CodeLeom**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by zolomohan** [comment]
Hi @kayere. If you haven't started with the revision yet, I would like to review this article. I noticed that you have a couple of articles on your plate.

**Comment by kayere** [comment]
Sorry @zolomohan, I have already started the review. I am almost done with the preliminary review. Otherwise, I would have assigned it to you. Regards.

**Comment by zolomohan** [comment]
That's fine. Thanks. Please go ahead.

File: articles/creating-a-virtual-assistant-using-python/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+author: aransiola-ayodele
+date: 2021-04-12T00:00:00-01:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/creating-a-virtual-assistant-using-python/hero.jpg
+    alt: Virtual Assistant Image Example
+---
+
+A virtual assistant, also known as an AI assistant or a digital assistant, is a computer program that recognizes natural language voice commands and performs tasks for the user. This article will use a python library to incorporate Alexa and will guide you through the process of creating your own voice assistant.
```
Comment (kayere) [inline articles/creating-a-virtual-assistant-using-python/index.md:17]:
```suggestion
A virtual assistant, also known as an AI assistant or a digital assistant, is a computer program that recognizes natural language voice commands and performs tasks for the user. This article guides you through the process of creating your own voice assistant through the use of a python library. 
```

File: articles/creating-a-virtual-assistant-using-python/index.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+![Python text to speech library download page](/engineering-education/creating-a-virtual-assistant-using-python/pyttsx3.png)
+
+Once you have successfully installed the second package, we will continue to install the third package called:
+
+3. Python Audio (PyAudio): this library provides Python bindings for PortAudio, the cross-platform audio I/O library. PyAudio allows you to quickly play and record audio on a number of devices using Python. Download the package [here](https://pypi.org/project/PyAudio/). Install the package using the steps used in 1 above.
+   ![Python Audio library download page](/engineering-education/creating-a-virtual-assistant-using-python/pyaudio.png)
+
+### Note:
+
+for Linux users, installing the Pyaudio, you might encounter an error, to fix the error and install PyAudio successfully, kindly follow the steps below:
```
Comment (kayere) [inline articles/creating-a-virtual-assistant-using-python/index.md:62]:
```suggestion
For Linux users, you might encounter an error when installing Pyaudio. Follow the following steps to fix the error:
```

File: articles/creating-a-virtual-assistant-using-python/index.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+sudo make install
+```
+
+6. Lastly, install pyaudio:
+
+```terminal
+pip install pyaudio
+```
+
+We have successfully installed the three libraries. Now, let get our Virtual assistant alive!!!
```
Comment (kayere) [inline articles/creating-a-virtual-assistant-using-python/index.md:96]:
```suggestion
We have successfully installed the three libraries. Now, let's get our Virtual assistant alive!!!
```

**Review by kayere — COMMENTED** [review]
Hello @CodeLeom, I am done with the preliminary review. I have noticed some issues with your grammar. Please go over the changes that I have suggested and review the article if possible to check on parts that I have missed.

**Comment by CodeLeom** [comment]
> Hello @CodeLeom, I am done with the preliminary review. I have noticed some issues with your grammar. Please go over the changes that I have suggested and review the article if possible to check on parts that I have missed.

Ok, I will revert with an updated changes. Thanks

**Review by kayere — APPROVED** [review]
@CodeLeom - Good work making the changes.

**Comment by hectorkambow** [comment]
@CodeLeom 


[Creating a virtual assistant using python #2008.pdf](https://github.com/section-engineering-education/engineering-education/files/6354589/Creating.a.virtual.assistant.using.python.2008.pdf)


👋 Good afternoon - 
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by CodeLeom** [comment]
> @CodeLeom
> 
> [Creating a virtual assistant using python #2008.pdf](https://github.com/section-engineering-education/engineering-education/files/6354589/Creating.a.virtual.assistant.using.python.2008.pdf)
> 
>  Good afternoon -
> Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.

Thanks for your feedback, I will revisit and make an update. Thanks

**Comment by CodeLeom** [comment]
@kayere plagiarism check is done. Thanks

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
#1870

**Pull request merged**